src/wp-includes/blocks.php

```diff
@@ -2356,3 +2356,46 @@ function _wp_footnotes_force_filtered_html_on_import_filter( $arg ) {
 	}
 	return $arg;
 }
+
+/**
+ * Returns the active block variation for a given block based on its attributes.
+ *
+ * Variations are determined by their `isActive` property, which is an array of
+ * block attribute keys that are compared to the given block's attributes using
+ * a strict equality check.
+ *
+ * If no `isActive` property is defined, all `attributes` specified by a variation
+ * are compared to the given block's to determine if the variation is active.
+ *
+ * @param WP_Block_Type $block_type       Block Type.
+ * @param array         $block_attributes Block attributes.
+ * @return array|null The active block variation, or null if no active variation is found.
+ */
+function get_active_block_variation( $block_type, $block_attributes ) {
+	foreach ( $block_type->get_variations() as $variation ) {
+		if (
+			isset( $variation['isActive'] ) &&
+			is_array( $variation['isActive'] ) &&
+			! empty( $variation['isActive'] )
+		) {
+			$attributes = $variation['isActive'];
+		} else if ( isset( $variation['attributes'] ) ) {
+			$attributes = array_keys( $variation['attributes'] );
+		} else {
+			// We have no way to determine if this is the active variation.
+			continue;
+		}
+
+		foreach ( $attributes as $attribute ) {
```

Comment:
I don't think a fallback like this can work well due to the nuances of `object` attributes handling. An example is `query` attribute of Query Loop block.

Reply:
Removed the fallback in https://github.com/WordPress/wordpress-develop/pull/6602/commits/9d54dd69d2e179a77b2e671b45a597c0dc3b6024.